src/StoreApi/docs/guiding-principles.md

```diff
@@ -0,0 +1,136 @@
+
+-   Properties should use snake_case 🐍
+-   Ambiguous terms should be avoided, and property names should try to use understandable language, rather than “WooCommerce” terminology or setting names
+-   Properties should be defined using US English, but the descriptions of fields should be localized
+-   Multiple types are permitted, for example, using a `null` type if a value is not applicable
+-   `sanitize_callback` and `validate_callback` are encouraged where possible to ensure data is received in the correct format before processing requests
+
+If you’re struggling to define a consistent schema, your approach may be flawed. A common real-world example of this would be representing something like _Post Tags_. It may be tempting to use the Slug as the property field name in the response:
+
+```php
+tags: [
+     "my-tag": {
+          // ...tag data
+     },
+     "my-other-tag": {
+          // ...tag data
+     }
+]
+```
+
+However, this is difficult to represent in Schema and is not predictable for the client. A better approach would be to use an array of data, with one of the properties being the Slug:
+
+```php
+tags: [
+     {
+          "slug": "my-tag",
+          // ...tag data
+     },
+     {
+          "slug": "my-other-tag",
+          // ...tag data
+     }
+]
+```
+
+## Routes should be designed around resources with a single type of schema
+
+Routes should be designed around resources (nouns) rather than operations (verbs). Routes should also return only one type of data defined by their Schema. For example:
+
+| Route                 | Resource type | Expected data               |
+| :-------------------- | :------------ | :-------------------------- |
+| `wc/store/cart`       | Cart          | A cart object               |
+| `wc/store/cart/items` | Cart Item     | A list of cart item objects |
+| `wc/store/products`   | Product       | A list of product objects   |
+| `wc/store/products/1` | Product       | A product object            |
+
+There are 2 notable exceptions to this rule in the Store API; _Errors_ and _Cart Operations_.
+
+### Error Handling
+
+Errors, including validation errors, should return an error response code (4xx or 5xx) and a [`WP_Error` object](https://developer.wordpress.org/reference/classes/wp_error/). The `AbstractRoute` class will handle the conversion of the `WP_Error` object into a valid JSON response.
+
+Error messages should be localized, but do not need to be written with language aimed at customers (clients should use the given error code to create customer-facing notices as needed).
+
+### Cart Operations
+
+Some endpoints are designed around operations to avoid client’s needing to make multiple round trips to the API. This is purely for convenience.
```

Comment:
```suggestion
Some endpoints are designed around operations to avoid clients needing to make multiple round trips to the API. This is purely for convenience.
```